sky/execution.py

```diff
@@ -924,3 +959,177 @@ def _maybe_translate_local_file_mounts_and_sync_up(task: task_lib.Task):
+            raise ValueError(
+                'Specifying ports in resources is not allowed. SkyServe will '
+                'use the port specified in the service section.')
+
+    task.set_resources(
+        requested_resources.copy(ports=[task.service.replica_port]))
+
+    with rich_utils.safe_status(
+            '[cyan]Registering service on the controller[/]'):
+        success = _register_service_name(service_name)
+        if not success:
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError(
+                    f'The service {service_name!r} is already running. '
+                    'Update service will be supported in the future. For now, '
+                    '`sky serve down` and then `sky serve up` again.')
+
+    controller_name = serve.SKY_SERVE_CONTROLLER_NAME
+    # TODO(tian): Probably run another sky.launch after we get the load balancer
+    # port from the controller? So we don't need to open so many ports here. Or,
+    # we should have a nginx traffic control to refuse any connection to the
+    # unregistered ports.
+    # TODO(tian): Probably choose the same cloud if replica cloud is specified?
+    controller_resources = controller_resources.copy(
+        ports=[serve.LOAD_BALANCER_PORT_RANGE])
+
+    _maybe_translate_local_file_mounts_and_sync_up(task, prefix='serve')
+
+    with tempfile.NamedTemporaryFile(
+            prefix=f'service-task-{service_name}-',
+            mode='w',
+    ) as service_file, tempfile.NamedTemporaryFile(
+            prefix=f'controller-task-{service_name}-',
+            mode='w',
+    ) as controller_file:
+        task_config = task.to_yaml_config()
+        common_utils.dump_yaml(service_file.name, task_config)
+        remote_task_yaml_path = (
+            serve.generate_remote_task_yaml_file_name(service_name))
+        controller_log_file = (
+            serve.generate_remote_controller_log_file_name(service_name))
+        vars_to_fill = {
+            'remote_task_yaml_path': remote_task_yaml_path,
+            'local_task_yaml_path': service_file.name,
+            'google_sdk_installation_commands':
+                gcp.GOOGLE_SDK_INSTALLATION_COMMAND,
+            'service_name': service_name,
+            'controller_log_file': controller_log_file,
+            'envs': _shared_controller_env_vars(),
+        }
+        backend_utils.fill_template(serve.CONTROLLER_TEMPLATE,
+                                    vars_to_fill,
+                                    output_path=controller_file.name)
+        # TODO(tian): Probably we should support customizable setup commands.
+        controller_task = task_lib.Task.from_yaml(controller_file.name)
+        controller_task.set_resources(controller_resources)
+
+        # Set this to modify default ray task CPU usage to custom value
+        # instead of default 0.5 vCPU. We need to set it to a smaller value
+        # to support a larger number of services.
```

Comment:
remnant

Reply:
This is not a remnant; we depend on this variable to determine the value of the vCPU we use. In `sky/backends/backend_utils.py`:
<img width="817" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/cab85b37-ba26-4c59-af1a-85323afe6c8b">

Reply:
Ah, the grammar creates a misunderstanding. How about "# Set service_name so the backend will know to modify"